gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -184,10 +184,13 @@ private void addDataNode(DiffEntry change) {
     if (checkFilePath(change.getNewPath(), NODE_FILE_DEPTH)) {
       Path nodeFilePath = new Path(this.repositoryDir, change.getNewPath());
       try {
-        Config config = loadNodeFileWithOverrides(nodeFilePath);
-        Class dataNodeClass = Class.forName(ConfigUtils.getString(config, FlowGraphConfigurationKeys.DATA_NODE_CLASS,
+        // Temporarily resolve the data node config when constructing it, but leave it unresolved after to allow overriding
+        Config unresolvedConfig = loadNodeFileWithOverrides(nodeFilePath);
+        Config resolvedConfig = unresolvedConfig.resolve();
```

Comment:
Can the resolution be pushed into the BaseDataNode class?

Reply:
I would have done that, but for example the fs URI check happens in `FileSystemDataNode`, not `BaseDataNode`. I did it this way to avoid needing to add this resolution into the constructor of every child data node.

Reply:
Can we add a new member variable resolvedConfig in addition to rawConfig and have resolvedConfig set inside BaseDataNode? the uri check can use the resolvedConfig?

Reply:
Sure, that seems like a good solution